### PR TITLE
Enforce TypeScript and Less code formatting with prettier.

### PR DESCRIPTION
**Pull request by newswangerd**
- Added `galaxyui/.prettierrc.yaml` to configure code formatting options
- Added `make dev/prettier` to format `.ts` and `.less` files in `galaxyui/src`
- Added prettier to node dependencies
- Added `make test/prettier` to check for unformatted code
- Added check during dev container start to verify that no unformatted code is added
- Updated contributing guide

#861

### CODE REVIEW

**Comment by newswangerd** [comment]
@cutwater I added it to `scripts/docker/dev/start-develop.sh` because that's where `make test/jslint` is. Should the prettier and JSLint both be added to the travis file?

**Comment by newswangerd** [comment]
We probably should move JS Lint out of the `start-develop.sh` because it makes it prevents the container from starting when there are linting errors, which is really annoying

**Comment by chouseknecht** [comment]
@newswangerd if you take jslint out of the startup script, than you have to add it somewhere in the Travis script.

**Comment by newswangerd** [comment]
@chouseknecht that's what I had in mind. I'm testing that now

File: CONTRIBUTING.rst [hunk 1]
```diff
@@ -432,6 +432,19 @@ To run lint checks against JavaScript/TypeScript sources, execute:
...
+
+We use prettier to enforce code formatting for all of our TypeScript and less files.
+To automatically format your Angular code run:
+
+.. code-block:: console
+
+    $ make dev/prettier
+
+Unformatted code will cause the travis build to fail when you push your changes to
+GitHub.
```
Comment (chouseknecht) [inline CONTRIBUTING.rst]: We need to mention that there is a `.prettierrc.yaml`, where settings live.

Comment (chouseknecht) [reply CONTRIBUTING.rst]: Also, might be a good idea to add a link to https://prettier.io/docs/en/editors.html, so people have some idea of how to configure their editor.

**Pull request merged**